### DISCUSSION TITLE
bensql:  A new native cli talk with databend

### DISCUSSION THREAD

**Discussion by sundy-li** [databendlabs/databend#10841] [Show and tell]
We're excited to announce the release of [bendsql](https://crates.io/crates/bendsql)! 

`bendsql` is a native command-line tool for interacting with databend. You can use `bendsql` to connect to a databend server, execute queries, and view query results.

To get started with `bendsql`, simply install it using Cargo:

```
cargo install bendsql@0.1.1
```

Once connected, you can execute SQL queries using the CLI's REPL interface or by piping SQL commands to bendsql via stdin.

If you are too familiar with MySQL, you can set `alias mysql=bendsql`  :) 

`bendsql` also includes basic ***SQL keyword highlighting*** and ***auto-completion***, as well as support for SELECT queries. We're still working on adding more features, so stay tuned for updates!

<img width="1430" alt="image" src="https://user-images.githubusercontent.com/3325189/228732955-f3b1bcb2-71dd-4a8c-82a9-5b0068fafdde.png">

If you're interested in contributing to `bendsql` or have any feedback, please check out our [GitHub repository]( https://github.com/datafuselabs/databend-client/tree/main/cli) and [issues](https://github.com/datafuselabs/databend-client/issues/21)

multi-line SQL example:

```
❯ bendsql -h arch -u sundy -p abc
Welcome to bendsql.
Connecting to http://arch:8900/ as user sundy.

arch :) select 3\
+ 3 \
+ 45;

select 3+ 3 + 45;

┌────────────────┐
│ ((3 + 3) + 45) │
│     UInt32     │
├────────────────┤
│             51 │
└────────────────┘

1 rows in set (0.034 sec)

```

We hope you find `bendsql` to be a useful tool for interacting with databend, and we look forward to hearing your feedback.

**Comment by l1t1** [comment]
compare to bendsql, which one is better、

**Reply by everpcpc** [reply]
We are planning to migrate features in BendSQL to databend-cli later.

**Reply by flaneur2020** [reply]
We have plans to rewrite bendsql in rust, and the experimental codes are contained in databend-cli.

Later, the cli tool in rust will become our official cli tool.

**Comment by l1t1** [comment]
will you release the binary file?

**Reply by everpcpc** [reply]
Yes, releasing binary files is on schedule.

**Comment by l1t1** [comment]
the name is a bit long, suggest use a shorter one like `dbsql`

**Comment by sundy-li** [comment]
Update: We just update a new crate,  `databend-cli` is renamed into `bendsql` now.  It has progress features now  in  https://github.com/datafuselabs/databend-client/pull/29
